### PR TITLE
Dedupe map code

### PR DESCRIPTION
**Pull request by nightblade9**
- Removed duplicate generation code
- Removed unused variable (`walkable`)
- Added tests for previously-untestable code!
- Fixed bad monster/item generation! (#44)

### CODE REVIEW

**Comment by nightblade9** [comment]
@NegativeScript this is an example of how we can extract code that was previously untestable and also test it. And I found/fixed a bug by writing unit tests, which was a bonus.

File: furusiya/model/maps/generators/dungeon_generator.py [hunk 1]
```diff
@@ -4,31 +4,30 @@
...
+    NUM_MONSTERS = (30, 40)
 
     def __init__(self, area_map):
         self._rooms = []
         self._area_map = area_map
         self._generate_rooms()
-        self._generate_objects()
+
+        map_generator.generate_monsters(self._area_map,
+            random.randint(DungeonGenerator.NUM_ROOMS[0], DungeonGenerator.NUM_ROOMS[1]))
```
Comment (NegativeScript) [inline furusiya/model/maps/generators/dungeon_generator.py]: How about using starred unpacking here? Less clutter.

File: furusiya/model/maps/generators/forest_generator.py [hunk 2]
```diff
@@ -5,30 +5,34 @@
...
+    NUM_ITEMS = (10, 20)
+    NUM_MONSTERS = (30, 40)
 
     def __init__(self, area_map):
         self._area_map = area_map
         self._generate_trees()
-        self._generate_objects()
+        
+        map_generator.generate_monsters(self._area_map,
+            random.randint(ForestGenerator.NUM_MONSTERS[0], ForestGenerator.NUM_MONSTERS[1]))
```
Comment (NegativeScript) [inline furusiya/model/maps/generators/forest_generator.py]: Ditto.

File: furusiya/model/maps/generators/dungeon_generator.py (hunk 1 above)
Comment (nightblade9) [reply furusiya/model/maps/generators/dungeon_generator.py]: You mean something like `min, max = DungeonGenerator.NUM_ROOMS` or did you mean something else?

Comment (nightblade9) [reply furusiya/model/maps/generators/dungeon_generator.py]: If you mean `random.randint(*DungeonGenerator.NUM_ROOMS)`, this doesn't seem to run for me. I get a syntax error. :thinking:

**Pull request merged**